curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-data-structures-by-building-a-shopping-cart/63eea817673c8e1c22927fa6.md

```diff
@@ -0,0 +1,309 @@
+---
+id: 63eea817673c8e1c22927fa6
+title: Step 19
+challengeType: 0
+dashedName: step-19
+---
+
+# --description--
+
+Use `const` and destructuring to extract `name` and `price` variables from `product`.
+
+# --hints--
+
+You should use destructuring to get the `name` and `price` variables.
```

Comment:
The input, `const {name, price} = products` works here. Same as `const {name, price} = product`. In my understanding, it should just be `const {name, price} = product`. 

I looked through the test and I can't figure out why it works that way too.